tensorboard/webapp/hparams/_redux/hparams_reducers.ts

```diff
@@ -87,7 +89,82 @@ const reducer: ActionReducer<HparamsState, Action> = createReducer(
         metrics: nextMetricFilters,
       },
     };
-  })
+  }),
+  on(actions.dashboardHparamColumnAdded, (state, {column, nextTo, side}) => {
+    const {dashboardDisplayedHparamColumns: oldColumns} = state;
+
+    let destinationIndex = oldColumns.length; // Default to append at end.
+    if (nextTo !== undefined && side !== undefined) {
+      const nextToIndex = oldColumns.findIndex(
+        (col) => col.name === nextTo.name
+      );
+      if (nextToIndex !== -1) {
+        destinationIndex = side === Side.RIGHT ? nextToIndex + 1 : nextToIndex;
+      }
+    }
+    let newColumn = {...column, enabled: true};
+    let newColumns = [...oldColumns];
+    newColumns.splice(destinationIndex, 0, newColumn);
+
+    return {
+      ...state,
+      dashboardDisplayedHparamColumns: newColumns,
+    };
+  }),
+  on(actions.dashboardHparamColumnRemoved, (state, {column}) => {
+    const newColumns = state.dashboardDisplayedHparamColumns.filter(
+      ({name}) => name !== column.name
+    );
+
+    return {
+      ...state,
+      dashboardDisplayedHparamColumns: newColumns,
+    };
+  }),
+  on(actions.dashboardHparamColumnToggled, (state, {column: toggledColumn}) => {
+    const newColumns = state.dashboardDisplayedHparamColumns.map((column) => {
+      if (column.name === toggledColumn.name) {
+        return {
+          ...column,
+          enabled: !toggledColumn.enabled,
+        };
+      } else {
+        return column;
+      }
```

Comment:
nit: the else clause here is redundant
```suggestion
      }
      return column;
```

Reply:
Done